sunbird_service_api_test/src/test/java/org/sunbird/integration/test/course/batch/CreateCourseBatchTest.java

```diff
@@ -0,0 +1,179 @@
+      "testCreateCourseBatchFailureInviteOnlyWithInvalidMentor";
+
+  public static final String TEST_NAME_CREATE_COURSE_BATCH_SUCCESS_INVITE_ONLY =
+      "testCreateCourseBatchSuccessInviteOnly";
+  public static final String TEST_NAME_CREATE_COURSE_BATCH_SUCCESS_INVITE_ONLY_WITH_CREATED_FOR =
+      "testCreateCourseBatchSuccessInviteOnlyWithCreatedFor";
+  public static final String TEST_NAME_CREATE_COURSE_BATCH_SUCCESS_INVITE_ONLY_WITH_MENTORS =
+      "testCreateCourseBatchSuccessInviteOnlyWithMentors";
+  public static final String TEST_NAME_CREATE_COURSE_BATCH_SUCCESS_OPEN =
+      "testCreateCourseBatchSuccessOpen";
+
+  public static final String TEMPLATE_DIR = "templates/course/batch/create";
+  private static final String TODAY_DATE = new SimpleDateFormat("yyyy-MM-dd").format(new Date());
+
+  private String getCreateCourseBatchUrl() {
+    return getLmsApiUriPath("/api/course/v1/batch/create", "/v1/course/batch/create");
+  }
+
+  @DataProvider(name = "createCourseBatchFailureDataProvider")
+  public Object[][] createCourseBatchFailureDataProvider() {
+
+    return new Object[][] {
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITHOUT_NAME, false, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITHOUT_COURSE_ID, false, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITHOUT_ENROLLMENTTYPE, false, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_INVALID_ENROLLMENTTYPE,
+        false,
+        HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITHOUT_START_DATE, false, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_PAST_START_DATE, false, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_INVALID_COURSE_ID, false, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_PAST_END_DATE, false, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_END_DATE_BEFORE_START_DATE,
+        false,
+        HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_INVALID_CREATED_FOR, true, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_INVALID_MENTOR, true, HttpStatus.BAD_REQUEST
+      }
+    };
+  }
+
+  @Test(dataProvider = "createCourseBatchFailureDataProvider")
+  @CitrusParameters({"testName", "isCourseIdRequired", "httpStatusCode"})
+  @CitrusTest
+  public void testCreateCourseBatchFailure(
+      String testName, boolean isCourseIdRequired, HttpStatus httpStatusCode) {
+    getTestCase().setName(testName);
```

Comment:
Can you move setting test name to BaseCitrusTestRunner for each perform*Test method?